Tasks/SonarQubePreBuild/SonarQubePreBuildImpl.ps1

```diff
@@ -90,21 +90,39 @@ function CreateCommandLineArgs
     return $sb.ToString();
 }
 
-function UpdateArgsForPullRequestAnalysis
+function UpdateArgsForPullRequestAnalysis($cmdLineArgs, $serviceEndpoint)
 {
-	param([string]$cmdLineArgs)
+    $prcaEnabled = GetTaskContextVariable "PullRequestSonarQubeCodeAnalysisEnabled"
+    if ($prcaEnabled -ieq "true")
+    {
+        if ($cmdLineArgs -and $cmdLineArgs.ToString().Contains("sonar.analysis.mode"))
+        {
+            throw "Error: sonar.analysis.mode seems to be set already. Please check the properties of SonarQube build tasks and try again."
+        }
 
-	$prcaEnabled = GetTaskContextVariable "PullRequestSonarQubeCodeAnalysisEnabled"
-	if ($prcaEnabled -ieq "true")
-	{
-		if ($cmdLineArgs -and $cmdLineArgs.ToString().Contains("sonar.analysis.mode"))
-		{
-			throw "Error: sonar.analysis.mode seems to be set already. Please check the properties of SonarQube build tasks and try again."
-		}
+        $sqServerVersion = GetSonarQubeServerVersion $serviceEndpoint.Url $serviceEndpoint.Authorization.Parameters.UserName $serviceEndpoint.Authorization.Parameters.Password
+        Write-Verbose "PullRequestSonarQubeCodeAnalysisEnabled is true, setting command line args for sonar-runner."
 
-		Write-Verbose "PullRequestSonarQubeCodeAnalysisEnabled is true, setting command line args for incremental mode for sonar-runner..."
+        if (!$sqServerVersion)
+        {
+            #we want to fail the build step if SonarQube server version isn't fetched
+            throw "Error: Unable to fetch SonarQube server version. Please make sure SonarQube server is reachable at $($serviceEndpoint.Url)"
+        }
```

Comment:
Currently the pre-build build step would fail with the above message which I feel is descriptive enough to let the user know of the issue. I will try to re-create this scenario and send a screenshot if possible...